File: NDOTestApp/NDOTravelExpenses-Step 6 VB/BusinessClasses/MileageAllowance.vb
Imports NDO
Imports System


<NDOPersistent()> _
Public Class MileageAllowance
    Inherits Expense

    ' Properties
    Public Overrides ReadOnly Property Amount() As Decimal
        Get
            Return (CDec(_milesDriven) * New Decimal(4, 0, 0, False, 1))
        End Get
    End Property

    Public Property MilesDriven() As Integer
        Get
            Return _milesDriven
        End Get
        Set(ByVal value As Integer)
            _milesDriven = value
        End Set
    End Property

    Public Overrides ReadOnly Property [Text]() As String
        Get
            Return "Mileage Allowance"
        End Get
    End Property


    ' Fields
    Private _milesDriven As Integer
End Class

File: Tutorial/NDOTestApp/NDOTravelExpenses-Step 6/BusinessClasses/MileageAllowance.cs
using System;
using NDO;
using System.ComponentModel;

namespace BusinessClasses
{
    [NDOPersistent, DisplayName("Milage Allowance")]
    public partial class MileageAllowance : Expense
    {
        int milesDriven;
        public int MilesDriven
        {
            get { return milesDriven; }
            set { milesDriven = value; }
        }
        public MileageAllowance()
        {
        }

        public override decimal Amount
        {
            //TODO: call accountant and get correct values ;-) 
            get { return (decimal)milesDriven * (decimal)0.4; }
        }

        public override string Text
        {
            get { return "Mileage Allowance"; }
        }
    }
}
